### PR TITLE
Refactored accelerator deliverables view into components

### PR DESCRIPTION
**Pull request by tommylau523**
This is a refactor similar to what we did for the applicant side.

### CODE REVIEW

**Comment by tommylau523** [comment]
* **#2958** <a href="https://app.graphite.dev/github/pr/terraware/terraware-web/2958?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @tommylau523 and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: src/scenes/AcceleratorRouter/Deliverables/ApproveDeliverableDialog.tsx [hunk 1]
```diff
@@ -17,6 +21,22 @@ export default function ApproveDeliverableDialog({
...
 }: ApproveDialogProps): JSX.Element {
   const theme = useTheme();
+  const { activeLocale } = useLocalization();
+
+  const defaultApproveMessages = useMemo(() => {
+    if (!activeLocale) {
+      return undefined;
+    }
+
+    switch (deliverableType) {
```
Comment (tommylau523) [inline src/scenes/AcceleratorRouter/Deliverables/ApproveDeliverableDialog.tsx:31]: One of the changes made here is to use different messages for approving questions deliverables. Currently it shows a message about approving species deliverable which seems inaccurate.

File: src/scenes/AcceleratorRouter/Deliverables/DeliverableView.tsx [hunk 2]
```diff
@@ -64,30 +72,51 @@ const DeliverableViewWrapper = () => {
...
+        disabled: deliverable.status === 'Needs Translation',
+      },
+      {
+        label: strings.formatString(strings.STATUS_WITH_STATUS, strings.NOT_NEEDED) as string,
+        value: 'not_needed',
+        disabled: deliverable.status === 'Not Needed',
+      },
+    ];
+
+    return deliverable.type === 'Species' && deliverable.status === 'Approved'
```
Comment (tommylau523) [inline src/scenes/AcceleratorRouter/Deliverables/DeliverableView.tsx:115]: In questions deliverables previously, there is no option to set deliverable to not-needed or needs-translation. This is changed to give those options, but we can take that away if that is intended.

File: src/components/AcceleratorDeliverableView/DeliverableCard.tsx [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+  }
+
+  return (
+    <>
+      {deliverable.type === 'Document' ? (
+        <DocumentDeliverableCard deliverable={deliverable} />
+      ) : deliverable.type === 'Questions' ? (
+        <QuestionsDeliverableCard deliverable={deliverable} />
+      ) : (
+        <SpeciesDeliverableCard deliverable={deliverable} />
```
Comment (nickgraz) [inline src/components/AcceleratorDeliverableView/DeliverableCard.tsx:20]: Maybe a `switch` or `if...elseif` would be a bit more readable here. Also if the type is unknown/unexpected, we should not render the species deliverable card.

**Review by nickgraz — APPROVED** [review]
Looks good!

**Pull request merged**